app/notification.go

```diff
@@ -333,8 +333,8 @@ func sendNotificationEmail(post *model.Post, user *model.User, channel *model.Ch
 	if *utils.Cfg.EmailSettings.EnableEmailBatching {
 		var sendBatched bool
 		if result := <-Srv.Store.Preference().Get(user.Id, model.PREFERENCE_CATEGORY_NOTIFICATIONS, model.PREFERENCE_NAME_EMAIL_INTERVAL); result.Err != nil {
-			// if the call fails, assume it hasn't been set and don't batch notifications for this user
-			sendBatched = false
+			// if the call fails, assume that the interval has not been explicitly set and batch the notifications
+			sendBatched = true
```

Comment:
I'll be the first to agree that it isn't ideal. The reason I didn't check the error is that it comes from the third-party [gorp](https://godoc.org/github.com/go-gorp/gorp#SelectOne) library, and Stores are meant to implementations of an interface, so it felt wrong to check for an sql-specific error when the implementation might not be an sql implementation. 

I don't know what the Golang way of doing this is, but it seems like this interface should be architected such that it returns a reliable, documented error code in this case, regardless of what the underlying implementation is.